Bump to 1.84

cuviper opened the pull request:
https://forge.rust-lang.org/release/process.html#bump-the-stable-version-number-t-6-days-friday-the-week-before

r? ghost

cuviper committed:
Bump to 1.84

cuviper commented:
@bors r+ rollup=never

bors commented:
:pushpin: Commit 734481cbdc178c58cd21a28fd07563abcceef9d2 has been approved by `cuviper`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=cuviper 734481cbdc178c58cd21a28fd07563abcceef9d2 -->
<!-- homu: {"type":"Approved","sha":"734481cbdc178c58cd21a28fd07563abcceef9d2","approver":"cuviper","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

cuviper commented:
@bors p=1

bors commented:
:hourglass: Testing commit 734481cbdc178c58cd21a28fd07563abcceef9d2 with merge 5dfdba2c9820ecf5aea0058f15eb9e78286fb54b...
<!-- homu: {"type":"BuildStarted","head_sha":"734481cbdc178c58cd21a28fd07563abcceef9d2","merge_sha":"5dfdba2c9820ecf5aea0058f15eb9e78286fb54b"} -->

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-18`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/11305943912/job/31446000367) [(plain)](https://github.com/rust-lang-ci/rust/commit/5dfdba2c9820ecf5aea0058f15eb9e78286fb54b/checks/31446000367/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
failures:

---- core::config::tests::download_ci_llvm stdout ----
Detected LLVM as non-available: running in CI and modified LLVM in this change
thread 'core::config::tests::download_ci_llvm' panicked at src/core/config/tests.rs:24:5:
assertion failed: parse("").llvm_from_ci
   0: rust_begin_unwind
             at /rustc/8c27a2ba6b21f3406a51118643080f0591949827/library/std/src/panicking.rs:662:5
   1: core::panicking::panic_fmt
             at /rustc/8c27a2ba6b21f3406a51118643080f0591949827/library/core/src/panicking.rs:74:14
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/11305943912/job/31446806524)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/11305943912/job/31446806524","builder_name":"checks-actions"} -->

cuviper commented:
@bors retry

(hoping that #131448 lands with its fix)

bors commented:
:hourglass: Testing commit 734481cbdc178c58cd21a28fd07563abcceef9d2 with merge 36ffc0839e964b64b21193147980892063a42c30...
<!-- homu: {"type":"BuildStarted","head_sha":"734481cbdc178c58cd21a28fd07563abcceef9d2","merge_sha":"36ffc0839e964b64b21193147980892063a42c30"} -->

rust-log-analyzer commented:
The job **`aarch64-apple`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/11312081099/job/31459077128) [(plain)](https://github.com/rust-lang-ci/rust/commit/36ffc0839e964b64b21193147980892063a42c30/checks/31459077128/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
---- [ui] tests/ui/cfg/assume-incomplete-release/assume-incomplete.rs#assume stdout ----

error in revision `assume`: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/stage2/bin/rustc" "/Users/runner/work/rust/rust/tests/ui/cfg/assume-incomplete-release/assume-incomplete.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/Users/runner/.cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/Users/runner/work/rust/rust/vendor" "--sysroot" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/stage2" "--target=aarch64-apple-darwin" "--cfg" "assume" "--check-cfg" "cfg(FALSE,assume,no_assume)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/test/ui/cfg/assume-incomplete-release/assume-incomplete.assume/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/Users/runner/work/rust/rust/build/aarch64-apple-darwin/native/rust-test-helpers" "-L" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/test/ui/cfg/assume-incomplete-release/assume-incomplete.assume/auxiliary" "-Z" "assume-incomplete-release"
--- stderr -------------------------------
error[E0425]: cannot find function `foo_1` in this scope
##[error]  --> /Users/runner/work/rust/rust/tests/ui/cfg/assume-incomplete-release/assume-incomplete.rs:34:5
   |
   |
LL | fn foo_2() { }
   | ---------- similarly named function `foo_2` defined here
LL |     foo_1();
   |     ^^^^^ help: a function with a similar name exists: `foo_2`

error: aborting due to 1 previous error
---
---- [ui] tests/ui/cfg/assume-incomplete-release/assume-incomplete.rs#no_assume stdout ----

error in revision `no_assume`: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/stage2/bin/rustc" "/Users/runner/work/rust/rust/tests/ui/cfg/assume-incomplete-release/assume-incomplete.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/Users/runner/.cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/Users/runner/work/rust/rust/vendor" "--sysroot" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/stage2" "--target=aarch64-apple-darwin" "--cfg" "no_assume" "--check-cfg" "cfg(FALSE,assume,no_assume)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/test/ui/cfg/assume-incomplete-release/assume-incomplete.no_assume/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/Users/runner/work/rust/rust/build/aarch64-apple-darwin/native/rust-test-helpers" "-L" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/test/ui/cfg/assume-incomplete-release/assume-incomplete.no_assume/auxiliary"
--- stderr -------------------------------
error[E0425]: cannot find function `foo` in this scope
##[error]  --> /Users/runner/work/rust/rust/tests/ui/cfg/assume-incomplete-release/assume-incomplete.rs:37:5
   |
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/11312081099/job/31459253803)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/11312081099/job/31459253803","builder_name":"checks-actions"} -->

RalfJung commented:
Hm, strange... seems like something is wrong with `ver_cfg_rel`?

cuviper commented:
It looks like `rust.download-rustc=if-unchanged` activated, not accounting for the version change. So `ver_cfg_rel("-1")` tried 1.83 per the new `src/version` = 1.84, but the downloaded compiler *was* 1.83 -- assumed incomplete.

slanterns commented:
Maybe that's because https://github.com/onur-ozkan/rust/commit/1090d8920c4a2804b1ba5d6ca45712ba1a21b29e#diff-5f5330cfcdb0a89b85ac3547b761c3a45c2534a85c4aaae8fea88c711a7a65b2R2738?

cuviper commented on src/bootstrap/src/core/config/config.rs:
```diff
@@ -2780,7 +2780,7 @@ impl Config {
             .args(["diff-index", "--quiet", &commit])
             .arg("--")
-            .args([self.src.join("compiler"), self.src.join("library")])
+            .args([self.src.join("compiler"), self.src.join("library"), self.src.join("version")])
```
This is the spot I found, but there may be multiple places that need to notice it.

slanterns replied:
Should it be `src/version`?

cuviper replied:
Bah, not that simple -- it's failing again:
https://github.com/rust-lang-ci/rust/actions/runs/11312606957/job/31460221821#step:26:2845

cuviper replied:
> Should it be `src/version`?

Ah, in my haste I thought `self.src` already included that. But it's late for me, so I'll check back tomorrow.

slanterns replied:
> > Should it be `src/version`?
> 
> Ah, in my haste I thought `self.src` already included that. But it's late for me, so I'll check back tomorrow.

Yeah, and in https://github.com/rust-lang/rust/pull/131560#issuecomment-2408867266 there is a more recent change (which might be more suspicious).

onur-ozkan replied:
Here is my [fix attempt](https://github.com/onur-ozkan/rust/commit/f492a6d307a1627ed6254b81c230b7b4785125d7), maybe you would like to try that here (or should I file a PR?).

RalfJung replied:
In general this shows that `download-rustc=if-unchanged` is very tricky to get right -- the rustc build has a ton of inputs, bootstrap assembles various environment variables, etc. This list will inevitably get out of date as some logic somewhere changes.

Given that almost every PR changes `compiler` or `library`, and that those are already in the files-to-track -- is this really worth it? The risk of getting it wrong is high, and I don't think I understand the benefits.

RalfJung replied:
I opened an issue to discuss this problem: https://github.com/rust-lang/rust/issues/131658.

My personal preference for now would be to disable this logic entirely, to remove any risk for the upcoming release.

onur-ozkan replied:
> Given that almost every PR changes compiler or library, and that those are already in the files-to-track -- is this really worth it? 

I haven't checked the exact numbers but we get fair amount of PRs that don’t touch the `compiler` or `library`. Each of those cuts the PR CI time for `x86_64-gnu-tools` in half and the same applies for a lot of runners in merge CI (if the PR isn’t rolled up with others that touch `compiler` or `library`). On top of that, with https://github.com/rust-lang/rust/pull/119899, library builds are going to take longer time. So, using `download-rustc` is perfect fit for the library profile. But first, we wanted to make it the default in CI to ensure it doesn’t break due to other changes later on.

> In general this shows that download-rustc=if-unchanged is very tricky to get right -- the rustc build has a ton of inputs, bootstrap assembles various environment variables, etc. This list will inevitably get out of date as some logic somewhere changes.

> The risk of getting it wrong is high, and I don't think I understand the benefits.

I don't see what's tricky about it and how there is high change of getting it wrong. We have a test that checks if "if-unchanged" logic acts as expected when there is change in "compiler" tree, adding more tests and coverage on that logic should make this experience quite stable IMO. Also, keep in mind that this option isn’t enabled for dist runners.

RalfJung replied:
> I don't see what's tricky about it and how there is high change of getting it wrong. 

The logic was fatally wrong and had to be fixed 2 times already within less than a week (by missing files or folders that have to be checked for changes). If that's not very clear evidence that this is very fragile, I don't know what is.

onur-ozkan replied:
> My personal preference for now would be to disable this logic entirely (i.e., never use the downloaded rustc on CI), to remove any risk for the upcoming release.

It's not used in dist runners.

> The logic was fatally wrong and had to be fixed 2 times already within less than a week (by missing files or folders that have to be checked for changes). If that's not very clear evidence that this is very fragile, I don't know what is.

It was never tested on CI before and I think this is quite normal. But sure, I understand your concern.

RalfJung replied:
> We have a test that checks if "if-unchanged" logic acts as expected when there is change in "compiler" tree, 

What you'd need is a test which checks that if there is any change outside the "compiler" tree, nothing about the compiler changes. That would ensure that the check is correct. But of course that's basically impossible -- and without such a check, any mistake in the "list of folders to check" will cause subtle breakage.

RalfJung replied:
> It's not used in dist runners.

I don't see how that's sufficient. It's not like the other runners are irrelevant.

> It was never tested on CI before and I think this is quite normal.

I think it i a symptom of fragile design. More robust alternatives are possible and I'd like to known whether they have been considered. See https://github.com/rust-lang/rust/issues/131658.

RalfJung replied:
Also, 3 days before a beta branch is a very bad time to land something where you think it is "normal" that it completely breaks CI for other PRs.

cuviper replied:
> Here is my [fix attempt](https://github.com/onur-ozkan/rust/commit/f492a6d307a1627ed6254b81c230b7b4785125d7), maybe you would like to try that here (or should I file a PR?).

It's tricky since this PR will cut off what goes into 1.83-beta -- which won't see any more version changes, but will see channel and stage0 changes. But I can cherry-pick it when I do that branch promotion, so I think for now I'll just add it here to get the release train on track. Thanks!

cuviper commented:
@bors try

bors commented:
:hourglass: Trying commit 027c93aa3a1b4cc6ac2f0d0ae5cddc2b7d5490b8 with merge 4dbbfea78dd81ab08e99192c17bcb79b7e59bb65...
<!-- homu: {"type":"TryBuildStarted","head_sha":"027c93aa3a1b4cc6ac2f0d0ae5cddc2b7d5490b8","merge_sha":"4dbbfea78dd81ab08e99192c17bcb79b7e59bb65"} -->

rust-log-analyzer commented:
The job **`aarch64-apple`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/11312606957/job/31460221821) [(plain)](https://github.com/rust-lang-ci/rust/commit/4dbbfea78dd81ab08e99192c17bcb79b7e59bb65/checks/31460221821/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Updating files:  98% (48954/49953)
Updating files:  99% (49454/49953)
Updating files: 100% (49953/49953)
Updating files: 100% (49953/49953), done.
Switched to a new branch 'try'
branch 'try' set up to track 'origin/try'.
[command]/opt/homebrew/bin/git log -1 --format=%H
4dbbfea78dd81ab08e99192c17bcb79b7e59bb65
##[group]Run src/ci/scripts/setup-environment.sh
src/ci/scripts/setup-environment.sh
---
file:.git/config remote.origin.url=https://github.com/rust-lang-ci/rust
file:.git/config remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
file:.git/config gc.auto=0
file:.git/config http.https://github.com/.extraheader=AUTHORIZATION: basic ***
file:.git/config branch.try.remote=origin
file:.git/config branch.try.merge=refs/heads/try
file:.git/config remote.upstream.fetch=+refs/heads/*:refs/remotes/upstream/*
file:.git/config submodule.library/backtrace.active=true
file:.git/config submodule.library/backtrace.url=https://github.com/rust-lang/backtrace-rs.git
file:.git/config submodule.library/stdarch.active=true
---
---- [ui] tests/ui/cfg/assume-incomplete-release/assume-incomplete.rs#assume stdout ----

error in revision `assume`: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/stage2/bin/rustc" "/Users/runner/work/rust/rust/tests/ui/cfg/assume-incomplete-release/assume-incomplete.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/Users/runner/.cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/Users/runner/work/rust/rust/vendor" "--sysroot" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/stage2" "--target=aarch64-apple-darwin" "--cfg" "assume" "--check-cfg" "cfg(FALSE,assume,no_assume)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/test/ui/cfg/assume-incomplete-release/assume-incomplete.assume/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/Users/runner/work/rust/rust/build/aarch64-apple-darwin/native/rust-test-helpers" "-L" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/test/ui/cfg/assume-incomplete-release/assume-incomplete.assume/auxiliary" "-Z" "assume-incomplete-release"
--- stderr -------------------------------
error[E0425]: cannot find function `foo_1` in this scope
##[error]  --> /Users/runner/work/rust/rust/tests/ui/cfg/assume-incomplete-release/assume-incomplete.rs:34:5
   |
   |
LL | fn foo_2() { }
   | ---------- similarly named function `foo_2` defined here
LL |     foo_1();
   |     ^^^^^ help: a function with a similar name exists: `foo_2`

error: aborting due to 1 previous error
---
---- [ui] tests/ui/cfg/assume-incomplete-release/assume-incomplete.rs#no_assume stdout ----

error in revision `no_assume`: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/stage2/bin/rustc" "/Users/runner/work/rust/rust/tests/ui/cfg/assume-incomplete-release/assume-incomplete.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/Users/runner/.cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/Users/runner/work/rust/rust/vendor" "--sysroot" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/stage2" "--target=aarch64-apple-darwin" "--cfg" "no_assume" "--check-cfg" "cfg(FALSE,assume,no_assume)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/test/ui/cfg/assume-incomplete-release/assume-incomplete.no_assume/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/Users/runner/work/rust/rust/build/aarch64-apple-darwin/native/rust-test-helpers" "-L" "/Users/runner/work/rust/rust/build/aarch64-apple-darwin/test/ui/cfg/assume-incomplete-release/assume-incomplete.no_assume/auxiliary"
--- stderr -------------------------------
error[E0425]: cannot find function `foo` in this scope
##[error]  --> /Users/runner/work/rust/rust/tests/ui/cfg/assume-incomplete-release/assume-incomplete.rs:37:5
   |
```

</details>

onur-ozkan committed:
check `stage0`, `channel` and `version` files for CI rustc

These files have important role for compiler builds, so include them
in the "if-unchanged" rustc logic.

Signed-off-by: onur-ozkan <work@onurozkan.dev>

cuviper commented:
@bors try

bors commented:
:hourglass: Trying commit 6e6cbdd2df0ae5a025c458d95a4c84df1b6af092 with merge 351e7d4c03aeaf7e5e888e4f158cb8e6aef2c435...
<!-- homu: {"type":"TryBuildStarted","head_sha":"6e6cbdd2df0ae5a025c458d95a4c84df1b6af092","merge_sha":"351e7d4c03aeaf7e5e888e4f158cb8e6aef2c435"} -->

bors commented:
:sunny: Try build successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/11315335357/job/31467319763)
Build commit: 351e7d4c03aeaf7e5e888e4f158cb8e6aef2c435 (`351e7d4c03aeaf7e5e888e4f158cb8e6aef2c435`)
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/11315335357/job/31467319763"},"merge_sha":"351e7d4c03aeaf7e5e888e4f158cb8e6aef2c435"} -->

cuviper commented:
Let's get this release train moving...

@bors r+ p=10

bors commented:
:pushpin: Commit 6e6cbdd2df0ae5a025c458d95a4c84df1b6af092 has been approved by `cuviper`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=cuviper 6e6cbdd2df0ae5a025c458d95a4c84df1b6af092 -->
<!-- homu: {"type":"Approved","sha":"6e6cbdd2df0ae5a025c458d95a4c84df1b6af092","approver":"cuviper","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 6e6cbdd2df0ae5a025c458d95a4c84df1b6af092 with merge 9cbaf4ecc1d42adbbcafd420aaea66ff2b78ce42...
<!-- homu: {"type":"BuildStarted","head_sha":"6e6cbdd2df0ae5a025c458d95a4c84df1b6af092","merge_sha":"9cbaf4ecc1d42adbbcafd420aaea66ff2b78ce42"} -->

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/11316000886/job/31469155544)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/11316000886/job/31469155544","builder_name":"checks-actions"} -->

cuviper commented:
@bors retry

bors commented:
:hourglass: Testing commit 6e6cbdd2df0ae5a025c458d95a4c84df1b6af092 with merge 27861c429af736ea3a6bb015956c7286071b286d...
<!-- homu: {"type":"BuildStarted","head_sha":"6e6cbdd2df0ae5a025c458d95a4c84df1b6af092","merge_sha":"27861c429af736ea3a6bb015956c7286071b286d"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/11317347372/job/31472636814)
Approved by: cuviper
Pushing 27861c429af736ea3a6bb015956c7286071b286d to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"cuviper","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/11317347372/job/31472636814"},"merge_sha":"27861c429af736ea3a6bb015956c7286071b286d"} -->

rust-timer commented:
Finished benchmarking commit (27861c429af736ea3a6bb015956c7286071b286d): [comparison URL](https://perf.rust-lang.org/compare.html?start=36780360b62320a61e2234b17ec600e8e4785509&end=27861c429af736ea3a6bb015956c7286071b286d&stat=instructions:u).

### Overall result: ❌ regressions - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=36780360b62320a61e2234b17ec600e8e4785509&end=27861c429af736ea3a6bb015956c7286071b286d&stat=instructions:u)
This is the most reliable metric that we have; it was used to determine the overall result at the top of this comment. However, even this metric can sometimes exhibit noise.

|                                    | mean | range        | count |
|:----------------------------------:|:----:|:------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -    | -            | 0     |
| Regressions ❌ <br /> (secondary)  | 1.9% | [1.9%, 1.9%] | 1     |
| Improvements ✅ <br /> (primary)   | -    | -            | 0     |
| Improvements ✅ <br /> (secondary) | -    | -            | 0     |
| All ❌✅ (primary)                 | -    | -            | 0     |

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=36780360b62320a61e2234b17ec600e8e4785509&end=27861c429af736ea3a6bb015956c7286071b286d&stat=max-rss)
<details>
<summary>Results (primary -3.2%)</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -3.2% | [-3.2%, -3.2%] | 1     |
| Improvements ✅ <br /> (secondary) | -     | -              | 0     |
| All ❌✅ (primary)                 | -3.2% | [-3.2%, -3.2%] | 1     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=36780360b62320a61e2234b17ec600e8e4785509&end=27861c429af736ea3a6bb015956c7286071b286d&stat=cycles:u)
<details>
<summary>Results (secondary 2.8%)</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | 3.7%  | [2.1%, 5.3%]   | 6     |
| Improvements ✅ <br /> (primary)   | -     | -              | 0     |
| Improvements ✅ <br /> (secondary) | -2.9% | [-2.9%, -2.9%] | 1     |
| All ❌✅ (primary)                 | -     | -              | 0     |
</details>

### [Binary size](https://perf.rust-lang.org/compare.html?start=36780360b62320a61e2234b17ec600e8e4785509&end=27861c429af736ea3a6bb015956c7286071b286d&stat=size:linked_artifact)
This benchmark run did not return any relevant results for this metric.

**Bootstrap**: 781.427s -> 782.891s (0.19%)
**Artifact size**: 331.96 MiB -> 332.14 MiB (0.05%)
